fedimint-server/src/lib.rs

```diff
@@ -280,7 +280,8 @@ impl FedimintServer {
         // once we produce an outcome we no longer need to rejoin
         self.rejoin_at_epoch = None;
 
-        for epoch_num in self.next_epoch_to_process()..=last_outcome.epoch {
+        let next_epoch_to_process = self.next_epoch_to_process();
+        for epoch_num in next_epoch_to_process..=last_outcome.epoch {
```

Comment:
I don't understand yet why this is needed. https://users.rust-lang.org/t/i-dont-understand-this-lifetime-async-quirk/88883

Reply:
Fascinating problem!

Reply:
https://github.com/rust-lang/rust/issues/57017